### DISCUSSION TITLE
Cache size

### DISCUSSION THREAD

**Discussion by skeapskeap** [Krukov/cashews#141] [Q&A] [answered]
Is it possible to find out how many values are currently stored in cache? I found an in-memory cache only workaround. Unfortunatelly it's not gonna work with other types of backend since store attr is present only in backend: Memory.
```
from cashews import cache
from cashews.backends.memory import Memory

cache.setup("mem://")
backend: Memory = cache.tags_backend
size = len(backend.store)
```

**Comment by Krukov** [comment]
Hi

Unfortunately cashew do not have such api (get count of stored keys). Probably I can add it but for redis it will be hard to do it in a right way because in real world redis can be shared between applications and it is hard to understand which key was set by cashew and which not. 

Can you tell a little more about your use case?

**Reply by skeapskeap** [reply]
I use cachews for long term cache purpose (a few days). And that's why it's desirable to maintain metrics of how many items are stored in cache. Of course I might make my own package on top of redis but it seams quite labor-intensive since cachews already exists.

**Reply by Krukov** [reply]
Got you. I hope that you use prometheus for metrics - in this case  better option is to use[ redis-exporter ](https://github.com/oliver006/redis_exporter) . For memory cache - metrics [middleware](https://github.com/Krukov/cashews#middleware) . 

Any way I will add low level api that will provide number of keys in a storage

**Reply by Krukov** [reply, marked as answer]
I've just added the method `get_keys_count` 
```python
mem = cache.setup("mem://")
redis_backend = cache.setup("redis://")

await cache.get_keys_count() # total number of keys  in mem and redis backends
await redis_backend.get_keys_count()  # number of keys  in redis backend

```
